recipes/egl/system/conanfile.py

```diff
@@ -16,54 +18,24 @@ def configure(self):
-        if not pkg_config.provides:
-            raise ConanException("EGL development files aren't available, give up")
-        libs = [lib[2:] for lib in pkg_config.libs_only_l]
-        lib_dirs = [lib[2:] for lib in pkg_config.libs_only_L]
-        ldflags = [flag for flag in pkg_config.libs_only_other]
-        include_dirs = [include[2:] for include in pkg_config.cflags_only_I]
-        cflags = [flag for flag in pkg_config.cflags_only_other if not flag.startswith("-D")]
-        defines = [flag[2:] for flag in pkg_config.cflags_only_other if flag.startswith("-D")]
+    def system_requirements(self):
+        yum = Yum(self).install(["mesa-libEGL-devel"])
+        try:
+            # this works for ubuntu>=20, debian>=11, pop>=20
+            apt = Apt(self).install(["libegl-dev"])
+        except ConanException:
+            apt = Apt(self).install(["libegl1-mesa-dev"])
 
-        self.cpp_info.system_libs.extend(libs)
-        self.cpp_info.libdirs.extend(lib_dirs)
-        self.cpp_info.sharedlinkflags.extend(ldflags)
-        self.cpp_info.exelinkflags.extend(ldflags)
-        self.cpp_info.defines.extend(defines)
-        self.cpp_info.includedirs.extend(include_dirs)
-        self.cpp_info.cflags.extend(cflags)
-        self.cpp_info.cxxflags.extend(cflags)
+        pacman = PacMan(self).install(["libglvnd"])
+        zypper = Zypper(self).install(["Mesa-libEGL-devel"])
+        pkg = Pkg(self).install(["libglvnd"])
 
-    def system_requirements(self):
-        packages = []
-        if tools.os_info.is_linux and self.settings.os == "Linux":
-            if tools.os_info.with_yum:
-                packages = ["mesa-libEGL-devel"]
-            elif tools.os_info.with_apt:
-                ubuntu_20_or_later = tools.os_info.linux_distro == "ubuntu" and tools.os_info.os_version >= "20"
-                debian_11_or_later = tools.os_info.linux_distro == "debian" and tools.os_info.os_version >= "11"
-                pop_os_20_or_later = tools.os_info.linux_distro == "pop" and tools.os_info.os_version >= "20"
-                if ubuntu_20_or_later or debian_11_or_later or pop_os_20_or_later:
-                    packages = ["libegl-dev"]
-                else:
-                    packages = ["libegl1-mesa-dev"]
-            elif tools.os_info.with_pacman:
-                packages = ["libglvnd"]
-            elif tools.os_info.with_zypper:
-                packages = ["Mesa-libEGL-devel"]
-            else:
-                self.output.warn("Don't know how to install EGL for your distro.")
-        if tools.os_info.is_freebsd and self.settings.os == "FreeBSD":
-            packages = ["libglvnd"]
-        if packages:
-            package_tool = tools.SystemPackageTool(conanfile=self, default_mode='verify')
-            for p in packages:
-                package_tool.install(update=True, packages=p)
+        if all([True if result is None else False for result in [yum, apt, pacman, zypper, pkg]]):
```

Comment:
```suggestion
        if not any([yum, apt, pacman, zypper, pkg]):
```

Reply:
I think that check will not work if we have a numeric return for a package manager call...

Reply:
number? I thought it returned an object ....

Reply:
I suggest updating the documentation, https://docs.conan.io/en/latest/reference/conanfile/tools/system/package_manager.html#conan-tools-system-package-manager-apt-methods It's not clear what is returned. Also, methods are not detailed: https://docs.conan.io/en/latest/reference/conanfile/tools/system/package_manager.html#methods

Reply:
Yes, thanks for pointing that out, we have to update the docs about that. The return is the return code for the executed package manager command